File: new_message/src/msg_sub.cpp
#include "ros/ros.h"
#include "new_message/new_msg.h"
#include "iostream"
#include "sstream"

void callback(const new_message::new_msg::ConstPtr& msg){
    ROS_INFO("%s", msg->greeting.c_str());
    ROS_INFO("%d",msg->number);
}

int main(int argc, char **argv){
    ros::init(argc, argv, "msg_sub");
    ros::NodeHandle node;
    ros::Subscriber sub = node.subscribe("new_msg_chatter",10,callback);
    ros::spin();
    return 0;
}

File: new_message/scripts/msg_sub.py
import rospy
from new_message.msg import new_msg

def callback(data):
    print("%s = %d" %(data.greeting, data.number))

def new_msg_listener():
    sub = rospy.Subscriber("new_msg_chatter", new_msg, callback)
    rospy.init_node("new_msg_talker", anonymous=True)

    rospy.spin()

if __name__ == "__main__":
    try:
        new_msg_listener()
    except rospy.ROSInterruptException:
        pass